src/lib_json/json_reader.cpp

```diff
@@ -77,10 +78,8 @@ Features Features::strictMode() {
 // ////////////////////////////////
 
 bool Reader::containsNewLine(Reader::Location begin, Reader::Location end) {
-  for (; begin < end; ++begin)
-    if (*begin == '\n' || *begin == '\r')
-      return true;
-  return false;
+  return std::any_of(begin, end,
+                     [](const char& b) { return b == '\n' || b == '\r'; });
```

Comment:
`(auto b)`

Don't assume char unnecessarily and leave off the references.

Reply:
auto lambdas are a C++14 feature. This is a C++11 project.

Reply:
Oops, it's been a while. `char` then? Hey same # of letters.

Reply:
I'd keep the const. Compilers do interesting things with const. I assume the reference type is fine.

Reply:
Really `char` is enough to say what we want. I like code that's short and to the point. The rest of the type decorations are extra code to read. There's nothing interesting that a compiler is going to do with const or reference decorations here. Taking refs off of register-sized types can make them more efficient, and doesn't count as an odr-use.